.github/workflows/precompile-assets.yaml

```diff
@@ -0,0 +1,44 @@
+name: Precompile Assets
+on:
+  push:
+    branches:
+      - '**'
```

Comment:
Wondering if it might be better to run this action only on pushes on `master`. @ntkme @bartkamphorst any thoughts?

Reply:
Updated to `master` only and remove the pull_request trigger. - Technically it's possible to checkout `refs/pull/<id>/head` and commit to PR, but turns out `actions/checkout` will checkout PR as detached HEAD by default, so that will require some additional setup which might not be worth it.

Regarding master vs all branches, I think it does not matter that much, as the hash check is designed to only commit when assets actually change. - If we update a branch and then merge to master, it won't commit on master the second time as assets will not change at that time.

Reply:
I think we should run it both before and after, so that you will have test status on every single commit on master.  I added a trigger for doing that.

Reply:
Hmm, I think `docker-deploy` will now run twice -- once called from the `precompile-assets` step, and once from the push to `master`. Do you see any way to avoid this?

Additionally, I think we want to also precompile assets in the `docker-test` workflow, so that when a PR changes assets, the tests are ran against the compiled changes.

Thanks so much for your help! This will make it a lot easier to maintain our JS/CSS dependencies.

Reply:
Honestly, GitHub hosted runners are free so I don’t see running twice before and after having any kind of problem. It will give the commit history a check mark on every commit which I think is actually better.

Reply:
That's a fair point, although I was mostly thinking about double pushes to DockerHub. I'm not entirely sure how this works though; I guess maybe it won't re-push images with a digest that already exists on dockerhub?

We can also just merge as is, and adjust if necessary.

Reply:
Yes, docker digest will change given image content will be different. Also, docker tags are basically mutable pointers to digest.